### PR TITLE
Feature - Improved content edition

### PR DESCRIPTION
**Pull request by alcar**
#### What is the purpose of this pull request?

- Add a loading overlay to content editor;
- Add scope and condition information to configuration cards;
- Make component editor's header editable;
- Change the configuration card layout;
- Change the size and color of the component editor's headers' back arrow;
- Standardize the component editor's headers;
- Move configuration creation button up and change its label;
- Increase spacing between form buttons;
- Use `EditorContext`'s loading state instead of `FormMetaContext`'s one;
- Use method to get `EditorContext`'s loading state instead of an attribute;
- Fix `ModeButton` default messages;
- Fix `vtex.native-types` typings.

#### What problem is this solving?

Content edition bugs and UX issues.

#### How should this be manually tested?
Workspace: [https://featureimprovedcontentedition--storecomponents.myvtex.com/admin/cms/storefront/](https://featureimprovedcontentedition--storecomponents.myvtex.com/admin/cms/storefront/).

#### Screenshots or example usage
Before:

image

<hr>

After:

image

#### Types of changes
- Bug fix (a non-breaking change which fixes an issue)
- New feature (a non-breaking change which adds functionality)

### CODE REVIEW

File: react/components/EditorContainer/Sidebar/ConfigurationList/LayoutEditor.tsx [hunk 1]
```diff
@@ -140,12 +138,14 @@ class LayoutEditor extends Component<Props> {
       })
 
       formMeta.setWasModified(false, () => {
-        formMeta.toggleLoading(this.exit)
+        editor.setIsLoading(false)
+
+        this.exit
```
Comment (lbebber) [inline react/components/EditorContainer/Sidebar/ConfigurationList/LayoutEditor.tsx]: 👀

File: react/components/EditorContainer/Sidebar/ConfigurationList/List/Card/ConditionTags/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  title: string
+}
+
+const Tag: React.FunctionComponent<Props> = ({ kind, text, title }) => {
+  const iconByKind = React.useMemo(
+    () => ({
+      date: <CalendarIcon />,
+      utm: <PersonIcon />,
+    }),
+    []
```
Comment (lbebber) [inline react/components/EditorContainer/Sidebar/ConfigurationList/List/Card/ConditionTags/Tag.tsx]: 👀

File: react/components/EditorContainer/Sidebar/ConfigurationList/List/Card/index.tsx [hunk 3]
```diff
@@ -98,31 +129,51 @@ const Card = ({
...
         >
-          {text => <div className="i gray">{text}</div>}
+          {message => (
+            <div className="flex items-center">
+              {iconByScope[scope]}
+
+              <span className="ml2 f6">
+                {message}{' '}
+                <span className="fw5">
+                  {intl.formatMessage({
```
Comment (lbebber) [inline react/components/EditorContainer/Sidebar/ConfigurationList/List/Card/index.tsx]: ⛏ FormattedMessage

File: react/components/EditorContainer/Sidebar/ConfigurationList/List/Card/styles.css [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#action-menu-parent > div > div > div > button:active,
```
Comment (lbebber) [inline react/components/EditorContainer/Sidebar/ConfigurationList/List/Card/styles.css]: Add a comment explaining that this is (supposed to be) a temporary hack

File: react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx [hunk 5]
```diff
@@ -491,26 +462,34 @@ class ConfigurationList extends React.Component<Props, State> {
...
+        modal.close()
+      }
 
-        console.log(err)
+      this.props.showToast({
+        horizontalPosition: 'right',
+        message: 'Something went wrong. Please try again.',
       })
+
+      console.log(err)
```
Comment (lbebber) [inline react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx]: 🤔

File: react/components/EditorContainer/Sidebar/index.tsx [hunk 6]
```diff
@@ -50,9 +52,13 @@ const Sidebar: React.FunctionComponent<Props> = ({
         id="admin-sidebar"
         className="transition animated fadeIn b--light-silver bw1 z-2 h-100 pt8 pt0-ns calc--height-ns overflow-x-hidden w-100 font-display bg-white shadow-solid-x w-18em-ns admin-sidebar"
       >
-        <div className="h-100 flex flex-column dark-gray">
+        <div
+          className={`h-100 flex flex-column dark-gray ${
+            isLoading ? 'relative' : ''
```
Comment (lbebber) [inline react/components/EditorContainer/Sidebar/index.tsx]: add a TODO to see in the future whether this conditional `relative` is necessary

File: react/components/icons/TemplateIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+
+const TemplateIcon: React.FC = () => (
+  <div>
```
Comment (lbebber) [inline react/components/icons/TemplateIcon.tsx]: why does this icon includes a div?

File: react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx (hunk 5 above)
Comment (alcar) [reply react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx]: 😞

File: react/components/icons/TemplateIcon.tsx (hunk 7 above)
Comment (alcar) [reply react/components/icons/TemplateIcon.tsx]: I'll a comment next to it explaining why it's necessary.

**Pull request merged**